genomepy/annotation.py

```diff
@@ -0,0 +1,395 @@
+                        duplicate_contigs.append(line[matching_element])
+                    conversion_table[line[matching_element]] = line[0]
+        return conversion_table, list(set(duplicate_contigs))
+
+    def _conform_gtf(
+        self, conversion_dict: dict, filter_contigs: Optional[bool] = True
+    ):
+        """
+        Create an annotation gtf file with contigs conforming to the genome
+        Overwrites the existing gtf file.
+        Returns a list of contigs not present in the genome
+        """
+        missing_contigs = []
+        tmp_dir = mkdtemp(dir=self.genome_dir)
+        new_gtf_file = os.path.join(tmp_dir, os.path.basename(self.annotation_gtf_file))
+        with _open(self.annotation_gtf_file, "r") as old, _open(
+            new_gtf_file, "w"
+        ) as new:
+            for line in old:
+                splitline = line.split("\t")
+                if splitline[0] in conversion_dict:
+                    splitline[0] = conversion_dict[splitline[0]]
+                    line = "\t".join(splitline)
+                    new.write(line)
+                else:
+                    missing_contigs.append(splitline[0])
+                    if not filter_contigs:
+                        new.write(line)
+
+        os.replace(new_gtf_file, self.annotation_gtf_file)
+        rm_rf(tmp_dir)
+        return list(set(missing_contigs))
+
+    def bed_from_gtf(self):
+        """
+        Create an annotation bed file from an annotation gtf file.
+        Overwrites the existing bed file.
+        """
+        tmp_dir = mkdtemp(dir=self.genome_dir)
+
+        # unzip if needed
+        self.annotation_gtf_file, is_unzipped = gunzip_and_name(
+            self.annotation_gtf_file
+        )
+        new_bed_file = os.path.join(
+            tmp_dir, os.path.basename(re.sub(r"\.gz$", "", self.annotation_bed_file))
+        )
+
+        cmd = "gtfToGenePred {0} /dev/stdout | genePredToBed /dev/stdin {1}"
+        sp.check_call(cmd.format(self.annotation_gtf_file, new_bed_file), shell=True)
+
+        # gzip if needed
+        self.annotation_gtf_file = gzip_and_name(self.annotation_gtf_file, is_unzipped)
+        new_bed_file = gzip_and_name(
+            new_bed_file, self.annotation_bed_file.endswith(".gz")
+        )
+
+        os.replace(new_bed_file, self.annotation_bed_file)
+        rm_rf(tmp_dir)
+
+    def sanitize(self, filter_contigs: Optional[bool] = True):
+        """
+        Compares the genome and gene annotations.
+        If the contig names do not conform, attempt to fix this.
+        If the contig names do conform (after fixing), filter the annotation contigs for genome contigs.
+        Log the results in the README.
+
+        filter_contigs: remove contigs from the annotations that are missing from the genome?
+        """
+        if self.genome_file is None:
+            sys.stderr.write("A genome is required for sanitizing!")
+            return
+
+        if self._is_conforming():
```

Comment:
To me the `conforming` terminology is not immediately clear, what is meant by that?